src/Hit/Git/Status.hs

```diff
@@ -115,18 +115,19 @@ It also handles special case of binary files. Typical raw text returned by @git@
 can look like this:
 
 @
- .foo.un~    | Bin 0 -> 523 bytes
- README.md   |   4 ++++
- foo         |   1 +
- bar => baz  |   2 --
- qux => quux |   0
+ .foo.un~               | Bin 0 -> 523 bytes
+ README.md              |   4 ++++
+ foo                    |   1 +
+ test/{bar => foo/baz}  |   2 --
+ qux => quux            |   0
 @
 -}
 parseDiffStat :: [Text] -> Maybe DiffStat
 parseDiffStat = \case
     [diffStatFile, diffStatCount, diffStatSigns] -> Just DiffStat{..}
-    _:"=>":diffStatFile:diffStatCount:rest -> Just DiffStat
+    prevFile:"=>":newFile:diffStatCount:rest -> Just DiffStat
         { diffStatSigns = unwords rest
+        , diffStatFile = expandFilePath (unwords [prevFile, "=>", newFile])
```

Comment:
I see `unwords` here but the first thing `expandFilePath` does is splits given string by `=>`. I think it can make things simple if `expandFilePath` is given `(Text, Text)` -- a text before arrow and after. Or maybe something different can be done. The main idea is that once we already parsed the string and we have some information, we can use that information and avoid doing necessary work